README.md

```diff
@@ -1 +1,106 @@
+# iRODS Java HTTP API Wrapper
+
+Java client library designed and developed to wrap the new iRODS HTTP API. It provides java classes to cleanly interact 
+with all the functionality of the HTTP API and comes with tests and packaging.
+
+## How to Use?
+### Client
+Create a new `IrodsHttpClient` instance and pass in the base URL. 
+```java
+IrodsHttpClient client = new IrodsHttpClient("http://localhost:<port>/irods-http-api/<version>");
+```
+
+`<version>` must match the format, X.Y.Z, where X is the major version number, Y is the minor version number, and Z is the patch version number.
+
+You can only access the information endpoint without authenticating. All other operations require a token.
+
+### How to Use Operations
+All methods can be called by using a method chaining approach following this layout: 
+```java
+client.<endpoint>.<operation>(parameters);
+```
+On the client object, you chain the endpoint name (for example, collections), and then chain the operation name 
+(createCollection). 
+
+### What is being returned
+All methods return a `Response` object, which holds the HTTP status code and the body of the response (as JSON). Can be 
+retrieved by using `.getHttpStatusCode()` and `.getBody()`, respectively. You can parse the JSON response body using your 
+preferred JSON parsing library or method.
+
+### Optional Fields
+If the operation request contains one optional field, then that field will be of either `Optional` or `OptionalInt`. For 
+example, `setInheritance` in the collections endpoint has an `admin` field that takes an `OptionalInt`: 
+```java
+// Don't want to set the field: 
+client.collections.setInheritance("token", "lpath", 1, OptionalInt.empty());
+
+// Want to set the field to 1:
+client.collections.setInheritance("token", "lpath", 1, OptionalInt.of(1));
+```
+If the operation request includes more than one optional fields, the method will accept a params object that you pass as an 
+instance of a specific class containing those fields. You can set only the desired optional fields as needed. The name 
+of this class follows the pattern `endpointMethodParams`. If you do not need to pass any optional parameters, you can 
+simply pass `null`. For example, the `remove` operation in the collections endpoint has two optional fields: `noTrash` and `recurse`:
+
+
+```java
+// Want to set the noTrash field. Since remove() has more than one optional field, we will need to instantiate a class 
+// for that optional field.
+CollectionsRemoveParams removeParams = new CollectionsRemoveParams();
+removeParams.setNoTrash(1);
+client.collections().remove(token, lpath, removeParams);
+
+// Not setting any optional fields: 
+client.collections().remove(token, lpath, null);
+```
+
+
+
+## Usage Example 
+The following shows a demonstration on how to use the client to create and remove a collection.
+
+```java
+String baseUrl = "http://11.11.111.111:9000/irods-http-api/0.3.0";
```

Comment:
Replace the IP, port, and version with placeholders (e.g. `<host>:<port>`, etc).

Reply:
Done.